posthog/src/main/java/com/posthog/PostHog.kt

```diff
@@ -227,6 +242,23 @@ public class PostHog private constructor(
             getPreferences().setValue(DISTINCT_ID, value)
         }
 
+    private var isIdentified: Boolean
+        @JvmName("get-isIdentified")
+        get() {
+            var isIdentified: Boolean?
+            synchronized(identifiedLock) {
+                isIdentified = getPreferences().getValue(IS_IDENTIFIED) as? Boolean
```

Comment:
We can probably read this during the SDK init instead so we do it only once, and after use a in-memory flag to avoid reading from preferences every time.

Reply:
Yes I think we can do that

Reply:
Also, I don't think we need to do this, as `SharedPreferences` itself caches the values in memory using a Map.

Reply:
The `PostHogMemoryPreferences` does but the `PostHogSharedPreferences` does not really, indeed the Android `SharedPreferences`  does it but they have their internal locking which we don't have control so it's safer to have our flag I think, and it won't hurt anyway.

Reply:
I saw that the `PostHogSharedPreferences` is using Android's `SharedPreferences` under the hood, and what I've understood is that when the `setup` is done in the `Application.onCreate()`, the `PostHogSharedPreferences` is instantiated, and this is when `context.getPreferences()` is invoked and the preferences are cached into memory. Also, in my opinion, isn't it better to let the `SharedPreferences` handle it by default, because otherwise we would have to manually handle the invalidation/update of "memory prefs", e.g. on `identify()` or `reset()`. Please correct me if I'm wrong or I'm missing something.

Reply:
Nope you are not wrong, `SharedPreferences` does that, but we need an atomic operation with multiple instructions, the `SharedPreferences` is only locking the preferences itself, and our lock locks multiple things, for example, multiple threads calling `identify` in parallel might give a wrong state.

Reply:
Okay. How should I approach implementing this then?

Reply:
as you are doing already, using a var and a lock, `identifiedLock`.
the difference is that you will use a global var `isIdentified`, instead of local var, similar to `sessionId` and `sessionLock`.

Reply:
Can you please elaborate a bit what you mean by a global var and local var in this context?

Reply:
something like this https://github.com/PostHog/posthog-android/pull/160/commits/a7bf27aa3b83887c93ffdef517177396465c25de
https://www.enjoytutorials.com/kotlin-local-and-global-variables-tutorial/